### PR TITLE
feat(component): Campaign Banner

### PR DESCRIPTION
**Pull request by eugene-mitnik**
# Description

https://github.com/unimelb/homepage/issues/3

Fixes # 3 from homepage repository

Adds new CampaignBanner component

[PREVIEW](https://deploy-preview-1474--pattern-lib-unimelb.netlify.app/?selectedKind=Campaign&selectedStory=Campaign%20Banner%202021&full=0&addons=1&stories=1&panelRight=0&addonPanel=REACT_STORYBOOK%2Freadme%2Fpanel)

## Type of change

- [x] New feature (non-breaking change which adds functionality)

# Checklist:

- [x] Make sure to do not repeat yourself (DRY)
- [x] Snapshots tested
- [x] A11y tested
- [x] CSS utilises BEM naming convention and structure
- [x] Code does not mutate variables/objects/arrays but instead uses `map`, `filter` etc to return a new array/object
- [x] Crossbrowser testing (IE11, Safari 8+, iOS 8.4+, Android 4.4+, Firefox ESR (v52.x), iPhone (4s,6), iPad 2, Galaxy s5) 
- [x] My changes generate no new warnings
- [x] I have added tests (e2e and unit) that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [x] Added the new component to the index.js (Vue and Lib) export

# Reviewer Checklist

- [x] Code is not repeated (DRY)
- [x] ES6+ code is used where possible
- [x] Code does not mutate variables/objects/arrays but instead uses `map`, `filter` etc to return a new array/object
- [x] CSS utilises BEM naming convention and structure
- [x] Snapshots tested against a copy of dev snapshots to check UI changes are intended
- [x] A11y tested
- [x] Crossbrowser tested in at least IE11

### CODE REVIEW

File: components/campaign/CampaignBanner.vue [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+      </h1>
+      <p class="campaign-banner__text">
+        {{ text }}
+      </p>
+      <ButtonIcon
+        v-if="showButton"
+        class="btn--secondary"
+        :href="href"
+        size="sml">
+        LEARN MORE
```
Comment (gepifanio) [inline components/campaign/CampaignBanner.vue:28]: Please add this button text as prop

File: components/campaign/stories/CampaignBanner2021.vue [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+<script>
+import CampaignBanner from '../CampaignBanner.vue';
+import largeBackground from './static/1440w_campaign_banner.jpg';
+import mediumBackground from './static/768w_campaign_banner.jpg';
+import smallBackground from './static/320w_campaign_banner.jpg';
+
+export default {
+  components: { CampaignBanner },
+  data: () => ({
```
Comment (gepifanio) [inline components/campaign/stories/CampaignBanner2021.vue:18]: Please add the readme object: 

`  readme: { source: true, html: true },`

and if the automated code generated in the documentation to use the component as `<campaign-banner></campaign-banner>` is not accurate, please add a custom readme to explain how to use it.

File: components/campaign/stories/index.js [hunk 3]
```diff
@@ -8,8 +8,12 @@ import {
 import Default from './Default.vue';
 import DarkBg from './DarkBg.vue';
 import BannerButtons from './BannerButtons.vue';
+import CampaignBanner2021 from './CampaignBanner2021.vue';
+import CampaignBanner2021NoButton from './CampaignBanner2021NoButton.vue';
```
Comment (gepifanio) [inline components/campaign/stories/index.js:12]: This story is broken, the `CampaignBanner2021NoButton.vue` file doesn't exist in there

**Comment by gepifanio** [comment]
Comment: Deploy Started

Building with commit 233a2119a8ea668ef808e171bfe0021becc39d35

https://app.netlify.com/sites/pattern-lib-unimelb/deploys/5f2cf0ab14241100074ff9b2

**Comment by eugene-mitnik** [comment]
Hi @gepifanio.
thanks
I've checked all of your comments and made changes in PR. Please review and let me know if something else is needed.

File: components/campaign/CampaignBanner.vue (hunk 1 above)
Comment (eugene-mitnik) [reply components/campaign/CampaignBanner.vue]: done

File: components/campaign/stories/index.js (hunk 3 above)
Comment (eugene-mitnik) [reply components/campaign/stories/index.js:12]: fixed

**Comment by jaredh-um** [comment]
Hi Eugene @eugene-mitnik 

Looking good. Are you able to make the following adjustments:

- [ ] adjust button text to use `text-transform: uppercase` (for the instance inside this component only)
- [ ] adjust breakpoint for mobile changeover for text and image to be at the `max-width: 600px` breakpoint
- [ ] remove fixed `height` properties from `campaign-banner` container to allow image to scale to full height as needed (to avoid clipping of the white rectangle) - (needs to be cleared for each breakpoint)

**Comment by eugene-mitnik** [comment]
all done

File: components/campaign/CampaignBanner.vue [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+      default: 'learn more',
+    },
+    showButton: {
+      type: Boolean,
+      default: false,
+    },
+  },
+  data: () => ({
+    breakpoints: {
+      low: 600,
```
Comment (jaredh-um) [inline components/campaign/CampaignBanner.vue:70]: adjust `low` value to 599 to match `--bp-max-x-mobile` so mobile changeover is synchronized

File: components/campaign/_campaign-banner.css [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+    font-family: var(--ff-normal);
+    color: rgb(var(--col-background-primary-white));
+    text-transform: uppercase;
+    font-size: .889rem;
+    line-height: 1.25;
+    width: 9.333rem;
+    margin-bottom: 1.111rem;
+  }
+
+  @media (--bp-max-x-mobile) {
```
Comment (gepifanio) [inline components/campaign/_campaign-banner.css:47]: Hi @eugene-mitnik,

please change the way to use the media queries, and put them inside the classes itself.

eg:

&__container {
 @media (--bp-max-x-mobile) {}
}

&__heading {
 @media (--bp-max-x-mobile) {}
}

Comment (eugene-mitnik) [reply components/campaign/_campaign-banner.css]: @gepifanio done

File: components/campaign/CampaignBanner.vue (hunk 4 above)
Comment (eugene-mitnik) [reply components/campaign/CampaignBanner.vue]: @jaredh-um done

**Comment by eugene-mitnik** [comment]
@gepifanio @jaredh-um unit tests have been added

**Pull request merged**